src/test/java/dev/snowdrop/release/services/CPaaSConfigUpdateServiceTest.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package dev.snowdrop.release.services;
+
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import dev.snowdrop.release.model.POM;
+import dev.snowdrop.release.model.Release;
+import dev.snowdrop.release.model.cpaas.product.CPaaSProduct;
+import dev.snowdrop.release.model.cpaas.product.CPaaSProductFile;
+import dev.snowdrop.release.model.cpaas.release.CPaaSAdvisory;
+import dev.snowdrop.release.model.cpaas.release.CPaaSPipelines;
+import dev.snowdrop.release.model.cpaas.release.CPaaSRelease;
+import dev.snowdrop.release.model.cpaas.release.CPaaSReleaseFile;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author <a href="antcosta@redhat.com">Antonio Costa</a>
+ */
+@QuarkusTest
+@TestProfile(TestProfiles.IntegrationTests.class)
+public class CPaaSConfigUpdateServiceTest {
+
+    private static final String RELEASE = "2.4.0";
+    private static final String PREVIOUS_RELEASE = "2.3.6";
+    private static final String PRODUCT_FILE_NAME = "product.yml";
+    private static final String RELEASE_FILE_NAME = "release.yml";
+    private static final YAMLMapper MAPPER = new YAMLMapper();
+
+    @Inject
+    CPaaSConfigUpdateService service;
+
+    @Inject
+    CPaaSReleaseFactory factory;
+
+    @Inject
+    GitService git;
+
+    @Test
+    public void parseProductConfig() throws Throwable {
```

Comment:
You're checking the golden path, you also need to check what happens when the inputs are wrong…

Reply:
Though, of course, that can wait for later.

Reply:
I changed the tests a bit but I don't think I added anything out of the golden path. I'd like to keep it as is for now (unless I have some mistake of course) while I check how could I implement those alternative tests.

Reply:
Included some wrong input tests for the cpaas configuration. thanks!